test/e2e/density.go

```diff
@@ -186,6 +209,108 @@ var _ = Describe("Density", func() {
 			// Tune the threshold for allowed failures.
 			badEvents := BadEvents(events)
 			Expect(badEvents).NotTo(BeNumerically(">", int(math.Floor(0.01*float64(totalPods)))))
+
+			if itArg.runLatencyTest {
+				Logf("Schedling additional Pods to measure startup latencies")
+				// Use service to assure even spread of pods between nodes
+				service := &api.Service{
+					ObjectMeta: api.ObjectMeta{
+						Name: "latency-service",
+					},
+					Spec: api.ServiceSpec{
+						Selector: map[string]string{"name": additionalRCName},
+						Ports: []api.ServicePort{
+							{
+								Name:       "portname1",
+								Port:       80,
+								TargetPort: util.NewIntOrStringFromString("service1"),
+							},
+						},
+					},
+				}
+				_, err = c.Services(ns).Create(service)
+				expectNoError(err)
+
+				gotPods, err := c.Pods(ns).List(labels.Everything(), fields.Everything())
+				expectNoError(err)
+
+				var latencies []PodStartupLatencyData
+				additionalNameStr := strconv.Itoa(minionCount) + "-" + string(util.NewUUID())
+				additionalRCName = "my-hostname-latency" + additionalNameStr
+				config = RCConfig{Client: c,
+					Image:        "gcr.io/google_containers/pause:go",
+					Name:         additionalRCName,
+					Namespace:    ns,
+					PollInterval: itArg.interval,
+					Replicas:     minionCount,
+				}
+
+				observedStartTime := make(map[string]util.Time, 0)
+				var wg sync.WaitGroup
+				wg.Add(1)
+				go func() {
+					defer GinkgoRecover()
+					watch, err := c.Pods(ns).Watch(labels.SelectorFromSet(labels.Set(map[string]string{"name": additionalRCName})), fields.Set{"status.phase": "Running"}.AsSelector(), gotPods.ResourceVersion)
```

Comment:
handrolling your watch is very error prone, you don't know when the server or some random router drops you connection, and what the server might be sending or not sending at that time. The only really safe way to handle this is to list watch, like our libraries do. Suggest something like:

```go
    done := make(chan struct{})
    stopCh := make(chan struct{})

    pods, controller := framework.NewInformer(
        createPodLWForRC(rc, client),
        &api.Pod{},
        time.Second*30,
        framework.ResourceEventHandlerFuncs{
            AddFunc: checkPods,
            DeleteFunc: checkPods,
        },
    )
    checkPods := func(obj interface{}) {
        if /* pods meets condition */ {
            close(stopCh)
            done <- struct{}{}
        }
    }
    go controller.Run(stopCh)
    obj := <- outputChan

func createPodLWForRC(rc, client) {
        return &cache.ListWatch{
            ListFunc: func() (runtime.Object, error) {
                return rm.kubeClient.Pods(api.NamespaceAll).List(rc.labels(), fields.Everything())
            },
            WatchFunc: func(rv string) (watch.Interface, error) {
                return rm.kubeClient.Pods(api.NamespaceAll).Watch(rc.labels(), fields.Everything(), rv)
            },
        },
}
```

You can also split the watch part out into a follow up pr so we have a basic test in and working